ros/src/waypoint_updater/waypoint_updater.py

```diff
@@ -128,33 +130,35 @@ def traffic_cb(self, msg):
                 else:
                     self.set_waypoint_velocity(self.all_wps.waypoints, i + self.wp_current_start, -1.0)
 
-            for j in range(i, i + 30):
+            for j in range(i, i + extra_brake_wps):
                 self.set_waypoint_velocity(self.all_wps.waypoints, j + self.wp_current_start, -1.0)
 
             self.stopping = True
 
+        num_resume_wps = 30
         if light_wp < 0:
             if (not self.resuming) and (self.stopping):
-                rospy.loginfo("Green light now, resume velocity")
+                rospy.logwarn("Green light now, resume velocity")
+                rospy.logwarn("Target velocity is %s", self.speed_limit)
                 self.stopping = False
                 self.resuming = True
-                self.resuming_point = self.wp_current_start+50
-                resuming_in_action = 50
-                #target_speed = self.original_wps.waypoints[self.wp_current_start+20].twist.twist.linear.x
-                target_speed = 100/9
-                rospy.loginfo("Target velocity is %s", target_speed)
-                own_speed = self.current_twist.twist.linear.x
-                rospy.loginfo("Own velocity is %s", own_speed)
+                self.resuming_point = self.wp_current_start + num_resume_wps
+
+                min_resume_speed = 3.0
+                dist_to_resume = self.distance(self.all_wps.waypoints, self.wp_current_start, self.resuming_point)
+                for i in range(num_resume_wps):
+                    delta = self.distance(self.all_wps.waypoints, self.wp_current_start, self.wp_current_start + i)
+                    new_speed = ((self.speed_limit - min_resume_speed)/dist_to_resume) * delta + min_resume_speed
+                    self.set_waypoint_velocity(self.all_wps.waypoints, i + self.wp_current_start, new_speed)
+
                 self.all_wps = self.original_wps
-                for i in range(resuming_in_action):
-                    rospy.loginfo("i is %s", i)
-                    #self.set_waypoint_velocity(self.all_wps.waypoints,i+self.wp_current_start,own_speed+(target_speed-own_speed)*(i+1)/(resuming_in_action))
-                    self.set_waypoint_velocity(self.all_wps.waypoints,i+self.wp_current_start,target_speed)
-                for i in range(110):
-                    self.set_waypoint_velocity(self.all_wps.waypoints,i+self.wp_current_start+resuming_in_action,target_speed)
+                num_waypoints = len(self.all_wps.waypoints)
+                for j in range(i, num_waypoints - num_resume_wps - 1):
+                    self.set_waypoint_velocity(self.all_wps.waypoints, (j+self.wp_current_start) % num_waypoints, self.speed_limit)
```

Comment:
C.f. the previous comment, now we don't need to set all the way to the end with speed_limit. Just need to make sure we change previously slowed velocity back.